src/AbstractWalkers/AtomWalkers.jl

```diff
@@ -9,79 +9,187 @@ The `AtomWalker` struct represents a walker composed of atoms/molecules.
 - `configuration::FastSystem`: The configuration of the walker.
 - `energy::typeof(0.0u"eV")`: The energy of the walker.
 - `iter::Int64`: The current iteration number of the walker.
-- `num_frozen_part::Int64`: The number of frozen particles in the walker.
+- `list_num_par::Vector{Int64}`: The list of the number of particles for each component.
+- `frozen::Vector{Bool}`: A boolean vector indicating whether each component is frozen or not.
 - `energy_frozen_part::typeof(0.0u"eV")`: The energy of the frozen particles in the walker, serves as a constant energy offset
     to the interacting part of the system.
 
 # Constructor
-```julia
-AtomWalker(configuration::FastSystem; energy=0.0u"eV", iter=0, num_frozen_part=0, energy_frozen_part=0.0u"eV")
-```
-Create a new `AtomWalker` with the given configuration and optional energy, iteration number, number of frozen particles, and energy of the frozen particles.
+- `AtomWalker(configuration::FastSystem; energy=0.0u"eV", iter=0, list_num_par=zeros(Int,C), frozen=zeros(Bool,C), energy_frozen_part=0.0u"eV")`: 
+    Constructs a new `AtomWalker` object with the given configuration and optional parameters.
 
 """
-mutable struct AtomWalker 
+mutable struct AtomWalker{C}
```

Comment:
What does the {C} mean?

Reply:
It means `AtomWalker` is a parametrized type, where is this case, `C` is the number of components.